--- overcloud_controller_pacemaker.pp.ORIG	2015-12-21 14:10:59.212976385 -0500
+++ overcloud_controller_pacemaker.pp	2015-12-21 14:11:19.056766615 -0500
@@ -489,6 +489,7 @@
       mon_initial_members => downcase(hiera('ceph_mon_initial_members'))
     }
     include ::ceph::profile::mon
+    include ::ceph::profile::rgw
   }
 
   if str2bool(hiera('enable_ceph_storage', 'false')) {
@@ -1610,6 +1611,15 @@
 
   }
 
+
+  if $ceph::profile::params::enable_rgw
+  {
+    exec { 'create_radosgw_keyring':
+      command => "/usr/bin/ceph auth get-or-create client.radosgw.gateway mon 'allow rwx' osd 'allow rwx' -o /etc/ceph/ceph.client.radosgw.gateway.keyring" ,
+      creates => "/etc/ceph/ceph.client.radosgw.gateway.keyring" ,
+    }
+  }
+
 } #END STEP 4
 
 # SERVICES INIT
