--- /dev/null	2015-12-21 11:06:33.720000000 -0500
+++ rgw.pp	2015-11-30 15:52:51.000000000 -0500
@@ -0,0 +1,14 @@
+class ceph::profile::rgw {
+  require ::ceph::profile::base
+
+  if $ceph::profile::params::enable_rgw
+  {
+    ceph::rgw { "radosgw.gateway":
+      user => $ceph::profile::params::rgw_user,
+      rgw_frontends => $ceph::profile::params::rgw_frontends ,
+      frontend_type => $ceph::profile::params::frontend_type ,
+    }
+   
+  }
+
+}
